sdk/src/utils/window.rs

```diff
@@ -37,23 +36,22 @@ pub struct WindowSize {
 /// }
 /// ```
 pub fn use_window_size() -> ReadOnlySignal<WindowSize> {
-    let mut window_size = use_signal(get_window_size);
-
-    // Initialize the handler
-    let tx = use_coroutine(|mut rx: UnboundedReceiver<WindowSize>| async move {
-        while let Some(data) = rx.next().await {
-            window_size.set(data);
+    let window_size = match try_use_context::<Signal<WindowSize>>() {
+        Some(w) => w,
+        // This should only run once.
+        None => {
+            let size = provide_root_context(Signal::new(get_window_size()));
```

Comment:
I am not entirely sure about this, you are creating a Signal under this scope and then injecting in to the root, so when this scope is dropped, the signal will be dropped as well but the context will still be there.

I think you should create the Signal directly into the Root Scope like I did in here https://github.com/marc2332/freya/pull/668/files#diff-56b3409ec38d3b001ac417e5c035ac192b65cdb63cfe48153bae812d6349e48aR79

Reply:
Ooo I did not know that `Signal`s were dropped based on the scope they were created. I also spotted another issue I think